### DISCUSSION TITLE
How to get data from join table?

### DISCUSSION THREAD

**Discussion by ghost** [supabase/supabase#7543] [Questions] [answered]
I have 3 tables 
listing, category and listing_category as join table, i was able to fetch data before but now i get null. 
`const fetchListing = async () => {
    let { data } = await supabase
      .from('listing')
      .select(`
    id,
    name,
    telephone,
    isSponsored,
    img,
    category(id,name)
    `)
    
    setCard(data);
    console.log(data);
  }`
 Can anyone help me about that?
Thank you

**Comment by steve-chavez** [comment, marked as answer]
Can you share the definition of `listing_category`? There was a change in how many-to-many relationships were detected. If `listing_category` has a `listing_id` and `category_id` they now must be included as part of `listing_category` primary key.

**Reply by ghost** [reply]
[
![Ekran Resmi 2022-07-03 23 07 03](https://user-images.githubusercontent.com/42077752/177055293-ee94b9da-d439-4da4-9f94-a21e62495745.png)
](url)

**Reply by steve-chavez** [reply]
You can make that the primary key consists of `id`, `listing_id`, `category_id` through the Table Editor UI.

Check the steps in this related discussion: https://github.com/supabase/supabase/discussions/7519#discussioncomment-3066861

**Reply by ghost** [reply]
Thank you the problem is solved now.
